### PR TITLE
noti service and fix i18n of api gateway

### PR DESCRIPTION
**Pull request by truongtamthanh2004**
# ✅ Checklist Tự review
## 🧹 Code sạch & format chuẩn
- [ ] Code sạch, dễ đọc, theo convention của team (naming, cấu trúc)
- [ ] Đã chạy `go fmt`, `goimports`, `golangci-lint` (nếu có)
- [ ] Không còn đoạn code tạm, `fmt.Println`, `log.Println`, comment debug
- [ ] Không còn biến, hàm, import không sử dụng
- [ ] Hàm không quá dài, có thể tách nhỏ nếu vượt quá 50 dòng
- [ ] Code không lặp lại, đã tái sử dụng hợp lý

## 🧠 Logic & Xử lý lỗi
- [ ] Logic đúng yêu cầu, không dư thừa, rõ ràng
- [ ] Xử lý đầy đủ các trường hợp `nil`, `error`, `invalid input`
- [ ] Mỗi thao tác có thể lỗi (I/O, DB, parse...) đều được check `err`
- [ ] Không để `err` bị silent — luôn log hoặc trả về (tùy ngữ cảnh)
- [ ] Sử dụng `context` đúng cách (nếu dùng), tránh leak
- [ ] Nếu là `goroutine`, có quản lý lifecycle và tránh leak

## 🌐 HTTP Handler (Gin)
- [ ] Sử dụng đúng HTTP method (GET, POST, PUT, DELETE…)
- [ ] Validate request (param, query, body) đầy đủ, rõ ràng
- [ ] Sử dụng binding (`ShouldBindJSON`, `ShouldBindUri`, v.v.) đúng cách
- [ ] Xử lý trả về `status code` phù hợp (`200`, `201`, `400`, `404`, `500`…)
- [ ] Response rõ ràng, có `message`, `data`, `error` (nếu cần)
- [ ] Không leak thông tin nhạy cảm trong response lỗi

## 🗃️ Struct & Layer
- [ ] Struct có annotation JSON đầy đủ (`json:"field_name"`)
- [ ] Sử dụng DTO/input/output rõ ràng, không lẫn lộn model DB và response
- [ ] Đã tách rõ các layer: handler → service → repository (nếu theo clean code)
- [ ] Middleware viết rõ ràng, dễ đọc, không làm chậm hệ thống

## 🧪 Unit Test & Mock
- [ ] Có test cho handler/service chính (ít nhất happy case và edge case)
- [ ] Nếu dùng DB/service ngoài, có mock hoặc test isolation
- [ ] Có test validate đầu vào (request, param, query)
- [ ] Không dùng test hardcode giá trị không ổn định (vd: time.Now() mà không mock)
- [ ] Tên test rõ ràng: `TestFunction_Scenario_Expected`

## 🔐 Security & Config
- [ ] Không hardcode secret, token, config trong code
- [ ] Không để lộ thông tin nhạy cảm qua log hoặc panic
- [ ] Validate và sanitize đầu vào người dùng
- [ ] Nếu có auth, kiểm tra middleware hoạt động đúng

## 🤖 Sử dụng GitHub Copilot
- [ ] Đã kiểm tra kỹ mọi đoạn code được gợi ý từ GitHub Copilot
- [ ] Không để lại phần comment hoặc TODO do Copilot sinh ra
- [ ] Không commit đoạn code mà chưa hiểu rõ logic hoặc mục đích
- [ ] Đảm bảo mọi function/code block từ Copilot phù hợp với context dự án
- [ ] Tối ưu lại code được gợi ý nếu cần — tránh lạm dụng hoặc copy nguyên mẫu

## Related Tickets
- ticket redmine link

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

### CODE REVIEW

**Comment by chienpv-3590** [comment]
commented

**Comment by truongtamthanh2004** [comment]
ready

File: api-gateway/internal/middleware/translate.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+				w.ResponseWriter.Header().Set("Content-Length", "")
+				w.ResponseWriter.WriteHeader(w.Status())
+				_ = json.NewEncoder(w.ResponseWriter).Encode(resp)
+				return
+			}
+		}
+
+		w.ResponseWriter.Header().Set("Content-Length", "")
+		w.ResponseWriter.WriteHeader(w.Status())
+		_, _ = w.ResponseWriter.Write(w.body.Bytes())
```
Comment (chienpv-3590) [inline api-gateway/internal/middleware/translate.go:45]: Hàm Write của responseBodyWriter chỉ ghi dữ liệu vào buffer mà không ghi ra response thật sự. Điều này dẫn đến việc response chỉ được ghi ra khi w.ResponseWriter.Write(w.body.Bytes()) ở cuối middleware. Tuy nhiên, điều này có thể làm mất một số metadata như cookies, headers set trong quá trình xử lý.

File: api-gateway/internal/middleware/translate.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+		c.Next()
+
+		if w.Status() >= 200 && w.Status() < 300 &&
+			strings.Contains(w.Header().Get("Content-Type"), "application/json") {
+
+			var resp map[string]interface{}
+			if err := json.Unmarshal(w.body.Bytes(), &resp); err == nil {
+				if msg, ok := resp["message"].(string); ok {
+					T := c.MustGet("T").(func(string) string)
```
Comment (chienpv-3590) [inline api-gateway/internal/middleware/translate.go:32]: Nếu không có key T, middleware sẽ panic. Nên kiểm tra kỹ hơn khi lấy ra hàm dịch.

File: services/notification-service/internal/handler/notification_handler.go [hunk 3]
```diff
@@ -1 +1,55 @@
...
+func (h *NotificationHandler) GetNotifications(c *gin.Context) {
+	userIDStr := c.Param("userId")
+	userID, err := strconv.Atoi(userIDStr)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"message": "notification.invalid_user_id"})
+		return
+	}
+	notifications, err := h.usecase.GetUserNotifications(uint(userID))
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"message": err.Error()})
```
Comment (chienpv-3590) [inline services/notification-service/internal/handler/notification_handler.go:51]: Trả về trực tiếp lỗi err.Error(), có thể lộ thông tin nhạy cảm. Nên chuẩn hóa message trả về cho client
